File: backend/app/core/schemas/comment.py
"""Comment schemas for API requests and responses."""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class CommentCreateRequest(BaseModel):
    """Request schema for creating a comment."""

    anchor_context: dict = Field(..., description="ProseMirror selection context")
    content: str = Field(..., min_length=1, max_length=1000)
    mentioned_user_ids: Optional[List[str]] = Field(None, description="Mentioned user IDs")


class CommentStatusUpdateRequest(BaseModel):
    """Request schema for updating comment status."""

    status: str = Field(..., pattern="^(open|resolved)$")


class CommentMessageResponse(BaseModel):
    """Response schema for a comment message."""

    user_id: str
    content: str
    created_at: datetime

    class Config:
        """Pydantic config."""

        from_attributes = True


class CommentResponse(BaseModel):
    """Response schema for a comment."""

    id: str
    document_id: str
    anchor_context: dict
    status: str
    mentioned_user_ids: List[str]
    messages: List[CommentMessageResponse]
    created_by: str
    created_at: datetime
    updated_at: datetime

    class Config:
        """Pydantic config."""

        from_attributes = True


class CommentListResponse(BaseModel):
    """Response schema for comment list."""

    comments: List[CommentResponse]
    total: int

File: frontend/src/services/api/comment.ts
import api from './client'
import { API_ENDPOINTS } from '../../config/api'

export interface Comment {
  id: string
  document_id: string
  content: string
  position?: {
    from: number
    to: number
  }
  created_at: string
  created_by: string
  status: 'open' | 'resolved'
  mentions?: string[]
}

export interface CommentListResponse {
  comments: Comment[]
  total: number
}

export const commentService = {
  // Get all comments for a document
  async getComments(documentId: string): Promise<CommentListResponse> {
    const response = await api.get(`${API_ENDPOINTS.DOCUMENTS.BASE}/comments/documents/${documentId}`)
    return response.data
  },

  // Create a new comment
  async createComment(
    documentId: string,
    content: string,
    position?: { from: number; to: number },
    mentions?: string[]
  ): Promise<Comment> {
    const response = await api.post(`${API_ENDPOINTS.DOCUMENTS.BASE}/comments/documents/${documentId}`, {
      content,
      position,
      mentions,
    })
    return response.data
  },

  // Update a comment
  async updateComment(
    commentId: string,
    content?: string,
    status?: 'open' | 'resolved'
  ): Promise<Comment> {
    const response = await api.put(`${API_ENDPOINTS.DOCUMENTS.BASE}/comments/${commentId}`, {
      content,
      status,
    })
    return response.data
  },

  // Delete a comment
  async deleteComment(commentId: string): Promise<void> {
    await api.delete(`${API_ENDPOINTS.DOCUMENTS.BASE}/comments/${commentId}`)
  },
}

